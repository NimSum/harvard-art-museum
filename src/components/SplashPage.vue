<template>
  <section class="SplashPage">
    <img v-for="image in images" v-bind:src="image.url" v-bind:key="image.id" alt="Random Art Photo"/>
  </section>
</template>

<script>
import getAnything from '../utils/apiFetches';
import apiUrls from '../utils/apiUrlGenerator';

export default {
  name: 'SplashPage',
  data() {
    return {
      images: [],
      next: ''
    }
  },
  async created() {
    const randomGalleryId = Math.floor(Math.random() * (18100 - 1 + 1)) + 1;
    const randomImagesUrl = apiUrls.randomImages(randomGalleryId);
    const result = await getAnything(randomImagesUrl);
    const images = result.records.map(record => ({
      url: record.baseimageurl,
      id: record.imageid
    }));
    this.$data.images = [...this.$data.images, ...images];
    this.$data.next = result.info.next;
  }
}
</script>

<style lang="scss" scoped>
  .SplashPage {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    padding: 1% 0;
    background: #08d9d6;
    height: 100%;
    overflow: scroll;

    img {
      display: inline-block;
      width: 27%;
      margin: 2%;
      transition: all 500ms ease-in-out;

      &:hover {
        width: 33%;
      }
    }
    
  }
</style>

