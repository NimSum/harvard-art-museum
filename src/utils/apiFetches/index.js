import axios from 'axios';

const getAnything = async (url) => {
  console.log('fetching')
  const response = await axios.get(url);
  if (response.statusText !== 'OK') {
    throw new Error('Failed to fetch data');
  }
  return response.data;
}

export default getAnything;