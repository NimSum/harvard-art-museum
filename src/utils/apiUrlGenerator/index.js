const apiUrls = {
  galleriesByFloor: (floor) => `https://api.harvardartmuseums.org/gallery?floor=${floor}&hasimage=1&apikey=${process.env.VUE_APP_API_KEY}&size=15&fields=galleryid,name,theme,objectcount,labeltext`,
  getGalleryObjects: (galleryId) => `https://api.harvardartmuseums.org/object?apikey=${process.env.VUE_APP_API_KEY}&sort=random&gallery=${galleryId}`,
  randomImages: (randomStart) => `https://api.harvardartmuseums.org/image?apikey=${process.env.VUE_APP_API_KEY}&size=21&sort=random&page=${randomStart}` 
}

export default apiUrls;