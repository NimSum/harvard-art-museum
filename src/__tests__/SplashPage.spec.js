import { mount } from '@vue/test-utils';
import SplashPage from '../components/SplashPage.vue';
import getAnything from '../utils/apiFetches';
import apiUrls from '../utils/apiUrlGenerator';
import { mockImages } from '../utils/mockData';

jest.mock('../utils/apiFetches')
getAnything.mockImplementation(() => Promise.resolve(mockImages));

describe('SplashPage', () => {
  let wrapper;
  const destroySpy = jest.fn();

  beforeEach(() => {
    wrapper = mount(SplashPage);
    jest.spyOn(wrapper.vm, 'addImages');
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });

  test('adds additional images from event', () => {
    const mockNewImage = {
      records: [
        {
          "id": 12345,
          "baseimageurl": "https://nrs.harvard.edu/urn-3:HUAM:INV028582_dynmc",
        }
      ]
    }
    getAnything.mockImplementation(() => Promise.resolve(mockNewImage));
    wrapper.vm.checkIfBottom();
    expect(wrapper.vm.addImages).toHaveBeenCalled();
  })
});