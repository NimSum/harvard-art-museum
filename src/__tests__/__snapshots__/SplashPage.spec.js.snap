// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage renders correctly 1`] = `
<section
  class="SplashPage"
>
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV028582_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV221494P_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV152006_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV185725_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:OCP19223_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV120600_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV115030_lgdl"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV165681_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:huam:INV19971503_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:VRS13982_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:21050_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:LEG3634_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV152372_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:71269_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV126374_lgdl"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV176191_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:INV139197_lgdl"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:COIN11049_dlvr"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:DDC108076_dynmc"
  />
  <img
    alt="Random Art Photo"
    src="https://nrs.harvard.edu/urn-3:HUAM:COIN00774_dynmc"
  />
</section>
`;
